service/gcs/main.go

```diff
@@ -18,8 +20,24 @@ import (
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
+	"golang.org/x/sys/unix"
 )
 
+func readMemoryEvents(efd uintptr, cgName string, limit int64) {
+	// Buffer must be >= 8 bytes for eventfd reads
+	// http://man7.org/linux/man-pages/man2/eventfd.2.html
+	buf := make([]byte, 8)
+	for {
+		_, err := cgrouputils.ReadEventFD(efd, buf)
+		if err != nil {
+			logrus.WithError(err).Warn("opengcs::readEvents - failed to read from eventfd")
+			unix.Close(int(efd))
+			return
+		}
+		logrus.Warnf("opengcs::readMemoryEvents - memory usage for cgroup '%s' exceeded %d byte threshold", cgName, limit)
```

Comment:
This seems sufficient for the gcs case, but for the container OOM case is there a way for us to get more detailed info?

Reply:
Even for GCS it would be good to get the memory use, perhaps the high water mark after a second or something?

Reply:
Sounds good

Reply:
@jstarks I was going to log mem-usage, mem-usage-limit, mem-usage-max and the same for swap and kernel (might as well if we have the stats), anything else you can think of that would be good to include? We have the whole suite of stats available so everything in this https://github.com/containerd/cgroups/blob/6c3dec43a1030a55584ab8b5b181411029d43ecc/stats/v1/metrics.pb.go#L279